website/scripts/build.js

```diff
@@ -72,45 +93,68 @@ async function build(previousFileSizes) {
 
     // Build the main website bundle
     const mainStats = await runWebpack(production({ browserWarningPath }));
-    handleErrors(mainStats);
-
-    console.log(chalk.green('Compiled successfully.'));
-    console.log();
+    handleErrors(log, mainStats);
 
-    console.log('File sizes after gzip:');
-    console.log();
-    printFileSizesAfterBuild(mainStats, previousFileSizes, parts.PATHS.build);
-    console.log();
-
-    console.log(`The ${chalk.cyan(parts.PATHS.build)} folder is ready to be deployed.`);
-    console.log();
+    log(chalk.green('Compiled successfully.'));
+    log('File sizes after gzip:');
+    printFileSizesAfterBuild(mainStats, previousDistFileSizes, parts.PATHS.build);
+  } catch (err) {
+    printErrors(log, 'Failed to compile prod.', _.castArray(err));
+    throw err;
+  }
+}
 
-    // Build the timetable-only build for the export service
-    console.log(chalk.cyan('Creating timetable-only build...'));
-    fs.removeSync(parts.PATHS.buildTimetable);
-    console.log(`${parts.PATHS.buildTimetable} has been removed`);
-    console.log();
+/**
+ * Build the timetable-only build for the export service.
+ */
+async function buildTimetableOnly() {
+  const log = (...args) => console.log('timetable-only:', ...args);
+  try {
+    log(chalk.cyan('Creating build...'));
 
     const timetableOnlyStats = await runWebpack(timetableOnly);
-    handleErrors(timetableOnlyStats);
+    handleErrors(log, timetableOnlyStats);
 
-    console.log(chalk.green('Compiled successfully.'));
-    console.log();
+    log(chalk.green('Compiled timetable-only successfully.'));
   } catch (err) {
-    printErrors('Failed to compile.', [err]);
-    process.exit(1);
+    printErrors(log, 'Failed to compile timetable-only.', _.castArray(err));
+    throw err;
   }
 }
 
-// Write commit hash into `commit-hash.txt` for reference during deployment.
-function writeCommitHash() {
-  const { commitHash } = parts.appVersion();
-  // Sync filename with `scripts/promote-staging.sh`.
-  fs.writeFileSync(path.join(parts.PATHS.build, 'commit-hash.txt'), `${commitHash.slice(0, 7)}\n`);
+async function buildAll(previousDistFileSizes) {
+  const results = await Promise.allSettled([
+    buildProd(previousDistFileSizes),
+    buildTimetableOnly(),
+  ]);
+  const errors = results.filter((r) => r.status === 'rejected').map((r) => r.reason);
+  if (errors.length > 0) {
+    throw errors;
+  }
+}
+
+async function main() {
+  console.log('Building version', chalk.cyan(parts.appVersion().versionStr));
+
+  // First, read the current file sizes in build directory.
+  // This lets us display how much they changed later.
+  const previousDistFileSizes = await measureFileSizesBeforeBuild(parts.PATHS.build);
+
+  // Clear previous build
+  await fs.remove(parts.PATHS.build);
+  console.log(`${parts.PATHS.build} has been removed`);
+
+  try {
+    await buildAll(previousDistFileSizes);
+  } catch (e) {
```

Comment:
The version of Node we're using should support this syntax 

```suggestion
  } catch {
```